### ISSUE TITLE
SPHERE: polarimetric observations not always found

### ISSUE THREAD

**Issue by joolof** [astropy/astroquery#2189]
Hi,
I am getting a strange behavior when searching for ESO/SPHERE polarimetric observations, as in some cases they are not showing up. 
Below a minimal working example:

```import numpy as np
from astroquery.eso import Eso
eso = Eso()
eso.login("joolof")
query = eso.query_instrument('sphere', column_filters={'target':'HD120326', 'dp_tech':'%'})
print(np.unique(query['ProgId']))
``` 
will return
``` 
095.C-0487(A)
095.C-0607(A)
```
But, running the following:
```
query = eso.query_instrument('sphere', column_filters={'target':'HD120326', 'dp_tech':'POLARIMETRY%'})
print(np.unique(query['ProgId']))
```
will return:
```
0101.C-0128(D)
 095.C-0487(A)
```
So the program "0101.C-0128(D)" is missed when using the "Any" keyword.
 
According the help for sphere, dp_tech can be:
`[x] dp_tech: % (Any), IFU% (IFU), IMAGE% (Any IMAGE), IMAGE,CORONOGRAPHY  (IMAGE,CORONOGRAPHY), POLARIMETRY% (Any POLARIMETRY), POLARIMETRY,CORONOGRAPHY (POLARIMETRY,CORONOGRAPHY), SPECTRUM (SPECTRUM), SPECTRUM,CORONOGRAPHY (SPECTRUM,CORONOGRAPHY)`

So I thought the "ANY" should include "Any POLARIMETRY" as well

Now, for another star, running the following queries:
```
query = eso.query_instrument('sphere', column_filters={'target':'HD32297', 'dp_tech':'POLARIMETRY%'})
query = eso.query_instrument('sphere', column_filters={'target':'HD32297', 'dp_tech':'%'})
```
will return:
```
098.C-0686(B)
```
and
```
096.C-0640(A)
098.C-0686(B)
 60.A-9354(A)
```
respectively, so the polarimetric dataset (098.C-0686(B)) is retrieved by astroquery.

I am using astroquery version 0.4.3. I know for sure that this was working before (but I don't know since when the behavior is different unfortunately) and I am a bit clueless on how to avoid this. I run several queries but it does not seem very elegant.

Any help is appreciated, thanks a lot!

**Comment by keflavich** [comment]
I can reproduce your results, but I don't know what's going on.  Can you reproduce these with the ESO web interface?  This is either a problem on the ESO server/web interface side, or there's an implementation error in how astroquery is passing the "any" `%` along; if this used to work and now doesn't, then it's a server side issue: nothing has changed in astroquery.eso for a while.

**Comment by joolof** [comment]
Thanks for the quick reply.

Checking on the SPHERE ESO archive:
http://archive.eso.org/wdb/wdb/eso/sphere/form
searching for HD120326, and setting "DPR Tech" to "Any" (the default value), the polarimetric OB (0101.C-0128(D)) shows up in the results.

Interestingly, I also see another ProgId, 0101.C-0016(A), which was not retrieved by astroquery. This one only shows up with astroquery if I set `'dp_tech':'IFU%'`, so it is not only an issue for polarimetric observations.

I thought maybe something changed around P101, but after checking several stars observed in P101, P104, and P105, sometimes it works, sometimes it doesn't. I cannot really find out what is the difference (e.g., HD121617, observed in P101 works with '%' but HD36112 observed in P104 never shows up regardless if I put '%' or 'POLARIMETRY%' (progid 0104.C-0472(A)).

Thanks!

**Comment by joolof** [comment]
Embarrassing that I did not think of that, but the ESO Helpdesk quickly realized that the `row_limit` was the problem. Setting it to -1 solved the issue and they could recover all the program ID.
Thanks

**Comment by keflavich** [comment]
Ah!  This isn't so obvious, since there are only three `ProgID`s.  So the problem is that, with the default limit, there are too many row entries for the first few program IDs, so later ones are excluded, right?  I don't think there's anything we should change to the module, but this highlights another case where warning about truncated returns might be useful.

**Comment by almicol** [comment]
Hi Adam, 
I'm Alberto Micol and I'd like to join the conversation, from the ESO perspective.

Could you please clarify your last comment here above? Should that warning be implemented by astroquery or by each provider (ESO in this case)?

Note 1: The Eso astroquery uses WDB underneath, and WDB always returns a message informing the user if more records are available or not; two examples:  the first message is for a complete result set, the second one for an incomplete one:
> A total of 309 records were found matching the provided criteria. 
> A maximum of 200 records were found matching the provided criteria - **any remaining rows were ignored**. 
But I guess that is not taken into consideration by astroquery (a bit difficult to parse).

Note 2: I'd suggest that astroquery implements such check at the client level.
An example to explain what I mean: We at ESO implemented the following logic in TAP (table access protocol): if the user sets to N the max num of output records, then TAP queries for N+1 records: if N+1 are returned, then TAP presents the first N (as requested by the user) and warns the user that more records are available; if N or less are returned, no warning is provided. 
TAP in this case is the client, and the database is the server. 
That is, the server does not implement any special logic, it is the client that plays smart.

Could not astroquery implement the same logic? Is that what you actually meant?

Many thanks,
Alberto

**Comment by bsipocz** [comment]
@keflavich - we certainly need to better document `ROW_LIMIT`, currently it's mentioned in one code comment in the narrative docs, but no troubleshooting tips and tips are provided.  Also, we may want to reconsider changing the defaults to `-1` here and for other modules that are not expected to return millions of rows for a query? (or something more obvious than the current 50, maybe that's just numerology but if I get 100 or 1000 in a result, it's more suspicious to be rated than when I get 50).

@joolof - would a section about the row limits in the narrative docs have helped you, somewhere here: https://astroquery.readthedocs.io/en/latest/eso/eso.html

**Comment by keflavich** [comment]
@almicol this is an astroquery issue, not an ESO issue.  So, yes, we can expand how astroquery handles this.

> Could not astroquery implement the same logic? Is that what you actually meant?

pretty much yes.

**Comment by joolof** [comment]
> @joolof - would a section about the row limits in the narrative docs have helped you, somewhere here: https://astroquery.readthedocs.io/en/latest/eso/eso.html

I suppose it would have helped. The "issue" being that I had a script that worked as intended (see below), so when I saw this behavior I only checked the docstring of the function, not the docs ...

After a bit of digging I understand better what happened: I had a script to do the query and reduce the data automatically, and I had an astroquery config file with the the `ROW_LIMIT` set to `-1`. Recently I changed my laptop and only put the user login in the config file, not the `ROW_LIMIT`, so that is why I was so sure "it used to work" and quite confused about what was happening.
